vdr/ambassador.go

```diff
@@ -121,8 +122,18 @@ func (n *ambassador) handleCreateDIDDocument(document dag.SubscriberDocument, pr
 	if err != nil {
 		return err
 	}
-	err = n.keyStore.AddPublicKey(document.SigningKey().KeyID(), rawKey, document.SigningTime())
-	if err != nil {
+	// Since we use an in-memory DID store, DIDs are re-registered on every startup. However, the keystore is persistent
+	// so we have to check whether we already have the public key to avoiding error conditions.
+	// TODO: this implementation is quite naive since the public key might actually differ or have a different validation
+	// period, which isn't taken into consideration. Should be fixed as part of https://github.com/nuts-foundation/nuts-node/issues/109
+	// since then we don't need to do this check any more.
+	_, err = n.keyStore.GetPublicKey(document.SigningKey().KeyID(), document.SigningTime())
+	if err != nil && errors.Is(err, nutsCrypto.ErrKeyNotFound) {
+		err = n.keyStore.AddPublicKey(document.SigningKey().KeyID(), rawKey, document.SigningTime())
```

Comment:
Could we just ignore an error of type `ErrKeyAlreadyExists`?

Reply:
Much simpler, good idea!